### PR TITLE
Some dll fixes

### PR DESCRIPTION
**Pull request by Alexius08**
**Some fixes for AI behavior**

Wonder building
- AIs can't build the Oracle before human-controlled Greece spawns

Teching
- AI Holy Rome prioritizes Academia for Protestantism
- AI Mali prioritizes Education for University of Sankore
- AI Russia prioritizes Urban Planning for St. Basil's Cathedral
- Reduced AI's general likelihood of beelining exploration, optics and
geography

Other
- Fixed null reference error that prevents compilation of debug version of DLL

### CODE REVIEW

File: CvGameCoreDLL/CvPlayerAI.cpp [hunk 1]
```diff
@@ -5042,8 +5042,7 @@ TechTypes CvPlayerAI::AI_bestTech(int iMaxPathLength, bool bIgnoreCost, bool bAs
 									if (iI == MYTHOLOGY || iI == DIVINATION || iI == CEREMONY)
 										iValue /= 5;
 								case GREECE:
-									if (iI == THEOLOGY)
```
Comment (dguenms) [inline CvGameCoreDLL/CvPlayerAI.cpp]: Can we keep C++ style curly brackets (isolated line for opening and closing) please?

File: CvGameCoreDLL/CvPlayerAI.cpp [hunk 2]
```diff
@@ -4978,10 +4978,10 @@ TechTypes CvPlayerAI::AI_bestTech(int iMaxPathLength, bool bIgnoreCost, bool bAs
 									iValue /= 2;
 								}
 
-								/*if (iI == ASTRONOMY || iI == COMPASS) { //otherwise they tend to beeline
```
Comment (dguenms) [inline CvGameCoreDLL/CvPlayerAI.cpp:4981]: Have you checked if Spain/Portugal still make it to the new world reasonably fast?

**Review by dguenms — CHANGES_REQUESTED** [review]

File: CvGameCoreDLL/CvPlayerAI.cpp (hunk 2 above)
Comment (Alexius08) [reply CvGameCoreDLL/CvPlayerAI.cpp:4981]: I'll check with some Inca/Aztec test games. Spain and Portugal still have massive inclination for researching Cartography, Exploration & Optics (lines 5182 and 5289, respectively).

Comment (Alexius08) [reply CvGameCoreDLL/CvPlayerAI.cpp:4981]: I just posted the results of my test games in the forum. Should I make further changes to the tech priority code or is it fine for now?

Comment (Alexius08) [reply CvGameCoreDLL/CvPlayerAI.cpp:4981]: I made further tweaks. AI Spain seems to be consistently ahead in the race for the new world. Portugal, however, can't catch up, even after letting them build more settlers after getting the tech for their UU. They instead prefer to wait for galleons before sending settlers to Brazil.

**Review by dguenms — APPROVED** [review]

**Pull request merged**